### DISCUSSION TITLE
Сollaborative editor

### DISCUSSION THREAD

**Discussion by shahzod-davlatov** [hexlet-volunteers/runit#357] [Ideas]
Хочу предложить внедрить многопользовательское редактирование сниппета, используя библиотеку [Yjs](https://github.com/yjs/y-monaco/)

Общение происходит через webrtc, позже можно будет заменить на websocket. 

Сценарий такой:

1. Пользователь генерирует уникальную ссылку-приглашение
2. Другие пользователи переходят по ней и присоединяются к редактированию его сниппета

Так будет выглядеть начальный вариант. Позже нужно будет добавить кодовое слово и список ссылок-приглашений, который будет периодически очищаться

**Comment by fey** [comment]
Давайте попробуем какой-нибудь MVP вариант. ТОлько вот я думаю - а нужно нам бекенд дорабатывать, т.е. нужен ли список пользователей, которые могут редактировать сниппет? Наверное да. Либо ссылка должна действовать какое-то время, пока активна хотя бы одна сессия, куда пригласили

**Comment by shahzod-davlatov** [comment]
Я реализую MVP вариант с ссылкой приглашением, работающей пока активна хотя бы одна сессия
